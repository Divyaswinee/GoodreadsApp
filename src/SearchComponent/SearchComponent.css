.searchContainer {
	display: flex;
	flex: 1;
	width: 100%;
	justify-content: center;
}

.searchBoxCustomize {
	width: 50em;
	border-radius: 25px;
}
