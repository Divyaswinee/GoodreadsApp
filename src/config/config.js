module.exports = {
  APIKEY: 'vZt6PSQ52iaN3bFKvEwOlQ',
  searchAPIURL: 'https://www.goodreads.com/search/index.xml',
  showAPIURL: 'https://www.goodreads.com/book/show/'
};