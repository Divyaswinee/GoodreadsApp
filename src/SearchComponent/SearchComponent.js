import React, { Component } from "react";
import AsyncSelect from 'react-select/lib/Async';
import axios from 'axios';
import proxify from 'proxify-url';
import PropTypes from 'prop-types';
import { APIKEY, searchAPIURL } from '../config/config';
import './SearchComponent.css';

class SearchComponent extends Component {
	state = {
    searchText: ''
  };

  static propTypes = {
    onSearchCallback: PropTypes.func
  }

  getSearchResult = (searchTerm) => {
    const url = `${searchAPIURL}?key=${APIKEY}&q=${encodeURI(searchTerm)}&search=title`;
    const proxyUrl = proxify(url, { inputFormat: 'xml' });
    return new Promise(resolve => {
      axios.get(proxyUrl)
        .then(({ data }) => {
          return data;
        })
        .then(datatwo => {
          const searchResult = datatwo.query.results.GoodreadsResponse.search;
          const allResult = searchResult.results;
          const workResult = allResult && allResult.work && allResult.work.length > 0 && allResult.work.map((current) => {
            return {
              value: current['id']['content'],
              author: current['best_book']['author']['name'],
              label: current['best_book']['title'],
              bookName: current['best_book']['title'],
              URL: current['best_book']['image_url'],
              averageRating: current['average_rating'],
              id: current['id']['content']
            }
          })
          console.log('workResult ', workResult);
          resolve(workResult);
        })
        .catch(error => {
          console.log('Error', error.toString());
        });
    });
  };

  searchHandleChange = (value) => {
    this.setState({ searchText: value });
    if(value.length === 0) {
      return this.props.showDetailCallBack(false);
    }
    this.props.onSearchCallback(value);
  };

  render() {
  	return (
  		<div className="searchContainer">
  	  	<AsyncSelect
          className="searchBoxCustomize"
          value={this.state.searchText}
          onChange={this.searchHandleChange}
          placeholder="Search Book"
          loadOptions={this.getSearchResult}
          defaultOptions
          backspaceRemoves={true}
          cache={false}
        />
      </div>
  	);
  }
}

export default SearchComponent;