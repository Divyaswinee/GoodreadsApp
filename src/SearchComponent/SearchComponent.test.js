import React from 'react';
import Enzyme, { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import axios from 'axios';
import proxify from 'proxify-url';
import AsyncSelect from 'react-select/lib/Async';
import SearchComponent from './SearchComponent';
import { APIKEY, searchAPIURL } from '../config/config';

Enzyme.configure({ adapter: new Adapter() });

describe('SearchComponent', () => {
  const mockedEvent = { target: {}, preventDefault: jest.fn() };
	const component = shallow(<SearchComponent />);

	jest.mock('axios', () => {
	  const receivedData = {"query":{"count":1,"created":"2018-10-08T12:26:40Z","lang":"en-GB","results":{"GoodreadsResponse":{"Request":{"authentication":"true","key":"vZt6PSQ52iaN3bFKvEwOlQ","method":"search_index"},"search":{"query":"pirate","results-start":"1","results-end":"20","total-results":"12750","source":"Goodreads","query-time-seconds":"0.16","results":{"work":[{"id":{"type":"integer","content":"46636002"},"books_count":{"type":"integer","content":"12"},"ratings_count":{"type":"integer","content":"19166"},"text_reviews_count":{"type":"integer","content":"3518"},"original_publication_year":{"type":"integer","content":"2017"},"original_publication_month":{"type":"integer","content":"2"},"original_publication_day":{"type":"integer","content":"28"},"average_rating":"4.00","best_book":{"type":"Book","id":{"type":"integer","content":"33643994"},"title":"Daughter of the Pirate King (Daughter of the Pirate King, #1)","author":{"id":{"type":"integer","content":"14358948"},"name":"Tricia Levenseller"},"image_url":"https://images.gr-assets.com/books/1483132546m/33643994.jpg","small_image_url":"https://images.gr-assets.com/books/1483132546s/33643994.jpg"}},{"id":{"type":"integer","content":"6618328"},"books_count":{"type":"integer","content":"76"},"ratings_count":{"type":"integer","content":"36245"},"text_reviews_count":{"type":"integer","content":"3864"},"original_publication_year":{"type":"integer","content":"2009"},"original_publication_month":{"type":"integer","content":"11"},"original_publication_day":{"type":"integer","content":"24"},"average_rating":"3.43","best_book":{"type":"Book","id":{"type":"integer","content":"6428887"},"title":"Pirate Latitudes","author":{"id":{"type":"integer","content":"5194"},"name":"Michael Crichton"},"image_url":"https://images.gr-assets.com/books/1332304096m/6428887.jpg","small_image_url":"https://images.gr-assets.com/books/1332304096s/6428887.jpg"}},{"id":{"type":"integer","content":"1434148"},"books_count":{"type":"integer","content":"41"},"ratings_count":{"type":"integer","content":"14578"},"text_reviews_count":{"type":"integer","content":"733"},"original_publication_year":{"type":"integer","content":"2003"},"original_publication_month":{"type":"integer","content":"10"},"original_publication_day":{"type":"integer","content":"1"},"average_rating":"3.82","best_book":{"type":"Book","id":{"type":"integer","content":"143727"},"title":"Pirates!","author":{"id":{"type":"integer","content":"83085"},"name":"Celia Rees"},"image_url":"https://images.gr-assets.com/books/1316638198m/143727.jpg","small_image_url":"https://images.gr-assets.com/books/1316638198s/143727.jpg"}},{"id":{"type":"integer","content":"1878218"},"books_count":{"type":"integer","content":"18"},"ratings_count":{"type":"integer","content":"22932"},"text_reviews_count":{"type":"integer","content":"557"},"original_publication_year":{"type":"integer","content":"2003"},"original_publication_month":{"type":"integer","content":"9"},"original_publication_day":{"type":"integer","content":"1"},"average_rating":"4.07","best_book":{"type":"Book","id":{"type":"integer","content":"18005"},"title":"How I Became a Pirate","author":{"id":{"type":"integer","content":"10899"},"name":"Melinda Long"},"image_url":"https://s.gr-assets.com/assets/nophoto/book/111x148-bcc042a9c91a29c1d680899eff700a03.png","small_image_url":"https://s.gr-assets.com/assets/nophoto/book/50x75-a91bf249278a81aabab721ef782c4a74.png"}},{"id":{"type":"integer","content":"44223004"},"books_count":{"type":"integer","content":"13"},"ratings_count":{"type":"integer","content":"9693"},"text_reviews_count":{"type":"integer","content":"1278"},"original_publication_year":{"type":"integer","content":"2015"},"original_publication_month":{"type":"integer","content":"11"},"original_publication_day":{"type":"integer","content":"3"},"average_rating":"3.74","best_book":{"type":"Book","id":{"type":"integer","content":"24611888"},"title":"Thomas Jefferson and the Tripoli Pirates","author":{"id":{"type":"integer","content":"273823"},"name":"Brian Kilmeade"},"image_url":"https://images.gr-assets.com/books/1434569443m/24611888.jpg","small_image_url":"https://images.gr-assets.com/books/1434569443s/24611888.jpg"}},{"id":{"type":"integer","content":"18721148"},"books_count":{"type":"integer","content":"16"},"ratings_count":{"type":"integer","content":"26305"},"text_reviews_count":{"type":"integer","content":"1558"},"original_publication_year":{"type":"integer","content":"2012"},"original_publication_month":{"type":"integer","content":"1"},"original_publication_day":{"type":"integer","content":"13"},"average_rating":"4.24","best_book":{"type":"Book","id":{"type":"integer","content":"13415554"},"title":"The Assassin and the Pirate Lord (Throne of Glass, #0.1)","author":{"id":{"type":"integer","content":"3433047"},"name":"Sarah J. Maas"},"image_url":"https://images.gr-assets.com/books/1335782612m/13415554.jpg","small_image_url":"https://images.gr-assets.com/books/1335782612s/13415554.jpg"}},{"id":{"type":"integer","content":"481231"},"books_count":{"type":"integer","content":"3"},"ratings_count":{"type":"integer","content":"6765"},"text_reviews_count":{"type":"integer","content":"531"},"original_publication_year":{"type":"integer","content":"2007"},"original_publication_month":{"type":"integer","content":"5"},"original_publication_day":{"type":"integer","content":"1"},"average_rating":"3.76","best_book":{"type":"Book","id":{"type":"integer","content":"493051"},"title":"To Catch a Pirate","author":{"id":{"type":"integer","content":"275067"},"name":"Jade Parker"},"image_url":"https://images.gr-assets.com/books/1475187589m/493051.jpg","small_image_url":"https://images.gr-assets.com/books/1475187589s/493051.jpg"}},{"id":{"type":"integer","content":"13238987"},"books_count":{"type":"integer","content":"17"},"ratings_count":{"type":"integer","content":"5244"},"text_reviews_count":{"type":"integer","content":"769"},"original_publication_year":{"type":"integer","content":"2012"},"original_publication_month":{"type":"integer","content":"10"},"original_publication_day":{"type":"integer","content":"2"},"average_rating":"3.71","best_book":{"type":"Book","id":{"type":"integer","content":"13539171"},"title":"Pirate Cinema","author":{"id":{"type":"integer","content":"12581"},"name":"Cory Doctorow"},"image_url":"https://images.gr-assets.com/books/1337400827m/13539171.jpg","small_image_url":"https://images.gr-assets.com/books/1337400827s/13539171.jpg"}},{"id":{"type":"integer","content":"42710965"},"books_count":{"type":"integer","content":"14"},"ratings_count":{"type":"integer","content":"6336"},"text_reviews_count":{"type":"integer","content":"938"},"original_publication_year":{"type":"integer","content":"2015"},"original_publication_month":{"type":"integer","content":"6"},"original_publication_day":{"type":"integer","content":"16"},"average_rating":"3.98","best_book":{"type":"Book","id":{"type":"integer","content":"23164968"},"title":"Pirate Hunters: Treasure, Obsession, and the Search for a Legendary Pirate Ship","author":{"id":{"type":"integer","content":"6243"},"name":"Robert Kurson"},"image_url":"https://images.gr-assets.com/books/1432123218m/23164968.jpg","small_image_url":"https://images.gr-assets.com/books/1432123218s/23164968.jpg"}},{"id":{"type":"integer","content":"1353276"},"books_count":{"type":"integer","content":"24"},"ratings_count":{"type":"integer","content":"5915"},"text_reviews_count":{"type":"integer","content":"320"},"original_publication_year":{"type":"integer","content":"1978"},"original_publication_month":{"nil":"true","type":"integer"},"original_publication_day":{"nil":"true","type":"integer"},"average_rating":"3.71","best_book":{"type":"Book","id":{"type":"integer","content":"724994"},"title":"A Pirate's Love","author":{"id":{"type":"integer","content":"29542"},"name":"Johanna Lindsey"},"image_url":"https://images.gr-assets.com/books/1293624669m/724994.jpg","small_image_url":"https://images.gr-assets.com/books/1293624669s/724994.jpg"}},{"id":{"type":"integer","content":"41413665"},"books_count":{"type":"integer","content":"16"},"ratings_count":{"type":"integer","content":"2241"},"text_reviews_count":{"type":"integer","content":"498"},"original_publication_year":{"type":"integer","content":"2015"},"original_publication_month":{"type":"integer","content":"2"},"original_publication_day":{"type":"integer","content":"3"},"average_rating":"2.90","best_book":{"type":"Book","id":{"type":"integer","content":"22529221"},"title":"We Are Pirates","author":{"id":{"type":"integer","content":"7176"},"name":"Daniel Handler"},"image_url":"https://images.gr-assets.com/books/1404528866m/22529221.jpg","small_image_url":"https://images.gr-assets.com/books/1404528866s/22529221.jpg"}},{"id":{"type":"integer","content":"21383294"},"books_count":{"type":"integer","content":"8"},"ratings_count":{"type":"integer","content":"10193"},"text_reviews_count":{"type":"integer","content":"1157"},"original_publication_year":{"type":"integer","content":"2013"},"original_publication_month":{"type":"integer","content":"6"},"original_publication_day":{"type":"integer","content":"18"},"average_rating":"3.86","best_book":{"type":"Book","id":{"type":"integer","content":"15714476"},"title":"The Pirate's Wish (The Assassin's Curse, #2)","author":{"id":{"type":"integer","content":"5331983"},"name":"Cassandra Rose Clarke"},"image_url":"https://images.gr-assets.com/books/1352903412m/15714476.jpg","small_image_url":"https://images.gr-assets.com/books/1352903412s/15714476.jpg"}},{"id":{"type":"integer","content":"229324"},"books_count":{"type":"integer","content":"12"},"ratings_count":{"type":"integer","content":"8257"},"text_reviews_count":{"type":"integer","content":"208"},"original_publication_year":{"type":"integer","content":"2007"},"original_publication_month":{"nil":"true","type":"integer"},"original_publication_day":{"nil":"true","type":"integer"},"average_rating":"4.16","best_book":{"type":"Book","id":{"type":"integer","content":"236764"},"title":"Pirates Don't Change Diapers","author":{"id":{"type":"integer","content":"10899"},"name":"Melinda Long"},"image_url":"https://s.gr-assets.com/assets/nophoto/book/111x148-bcc042a9c91a29c1d680899eff700a03.png","small_image_url":"https://s.gr-assets.com/assets/nophoto/book/50x75-a91bf249278a81aabab721ef782c4a74.png"}},{"id":{"type":"integer","content":"55803833"},"books_count":{"type":"integer","content":"7"},"ratings_count":{"type":"integer","content":"7323"},"text_reviews_count":{"type":"integer","content":"1276"},"original_publication_year":{"type":"integer","content":"2018"},"original_publication_month":{"type":"integer","content":"2"},"original_publication_day":{"type":"integer","content":"27"},"average_rating":"4.21","best_book":{"type":"Book","id":{"type":"integer","content":"36682619"},"title":"Daughter of the Siren Queen (Daughter of the Pirate King, #2)","author":{"id":{"type":"integer","content":"14358948"},"name":"Tricia Levenseller"},"image_url":"https://images.gr-assets.com/books/1511933735m/36682619.jpg","small_image_url":"https://images.gr-assets.com/books/1511933735s/36682619.jpg"}},{"id":{"type":"integer","content":"45065759"},"books_count":{"type":"integer","content":"9"},"ratings_count":{"type":"integer","content":"3774"},"text_reviews_count":{"type":"integer","content":"686"},"original_publication_year":{"type":"integer","content":"2016"},"original_publication_month":{"type":"integer","content":"1"},"original_publication_day":{"type":"integer","content":"12"},"average_rating":"4.18","best_book":{"type":"Book","id":{"type":"integer","content":"25330544"},"title":"The Only Pirate at the Party","author":{"id":{"type":"integer","content":"6825040"},"name":"Lindsey Stirling"},"image_url":"https://images.gr-assets.com/books/1435029398m/25330544.jpg","small_image_url":"https://images.gr-assets.com/books/1435029398s/25330544.jpg"}},{"id":{"type":"integer","content":"633221"},"books_count":{"type":"integer","content":"16"},"ratings_count":{"type":"integer","content":"6829"},"text_reviews_count":{"type":"integer","content":"133"},"original_publication_year":{"type":"integer","content":"1990"},"original_publication_month":{"nil":"true","type":"integer"},"original_publication_day":{"nil":"true","type":"integer"},"average_rating":"3.89","best_book":{"type":"Book","id":{"type":"integer","content":"574676"},"title":"Sassinak (Planet Pirates, #1)","author":{"id":{"type":"integer","content":"26"},"name":"Anne McCaffrey"},"image_url":"https://images.gr-assets.com/books/1528916184m/574676.jpg","small_image_url":"https://images.gr-assets.com/books/1528916184s/574676.jpg"}},{"id":{"type":"integer","content":"448620"},"books_count":{"type":"integer","content":"48"},"ratings_count":{"type":"integer","content":"13975"},"text_reviews_count":{"type":"integer","content":"526"},"original_publication_year":{"type":"integer","content":"1992"},"original_publication_month":{"nil":"true","type":"integer"},"original_publication_day":{"nil":"true","type":"integer"},"average_rating":"3.89","best_book":{"type":"Book","id":{"type":"integer","content":"143717"},"title":"Pirates Past Noon (Magic Tree House, #4)","author":{"id":{"type":"integer","content":"578"},"name":"Mary Pope Osborne"},"image_url":"https://images.gr-assets.com/books/1320523952m/143717.jpg","small_image_url":"https://images.gr-assets.com/books/1320523952s/143717.jpg"}},{"id":{"type":"integer","content":"218224"},"books_count":{"type":"integer","content":"20"},"ratings_count":{"type":"integer","content":"3567"},"text_reviews_count":{"type":"integer","content":"462"},"original_publication_year":{"type":"integer","content":"2004"},"original_publication_month":{"nil":"true","type":"integer"},"original_publication_day":{"nil":"true","type":"integer"},"average_rating":"3.87","best_book":{"type":"Book","id":{"type":"integer","content":"225314"},"title":"The Pirates! In an Adventure with Scientists","author":{"id":{"type":"integer","content":"83082"},"name":"Gideon Defoe"},"image_url":"https://s.gr-assets.com/assets/nophoto/book/111x148-bcc042a9c91a29c1d680899eff700a03.png","small_image_url":"https://s.gr-assets.com/assets/nophoto/book/50x75-a91bf249278a81aabab721ef782c4a74.png"}},{"id":{"type":"integer","content":"754588"},"books_count":{"type":"integer","content":"10"},"ratings_count":{"type":"integer","content":"2014"},"text_reviews_count":{"type":"integer","content":"171"},"original_publication_year":{"type":"integer","content":"2003"},"original_publication_month":{"nil":"true","type":"integer"},"original_publication_day":{"nil":"true","type":"integer"},"average_rating":"3.73","best_book":{"type":"Book","id":{"type":"integer","content":"768526"},"title":"The Pirate Next Door (Regency Pirates, #1)","author":{"id":{"type":"integer","content":"28569"},"name":"Jennifer Ashley"},"image_url":"https://s.gr-assets.com/assets/nophoto/book/111x148-bcc042a9c91a29c1d680899eff700a03.png","small_image_url":"https://s.gr-assets.com/assets/nophoto/book/50x75-a91bf249278a81aabab721ef782c4a74.png"}},{"id":{"type":"integer","content":"49328714"},"books_count":{"type":"integer","content":"18"},"ratings_count":{"type":"integer","content":"4362"},"text_reviews_count":{"type":"integer","content":"331"},"original_publication_year":{"type":"integer","content":"2016"},"original_publication_month":{"type":"integer","content":"9"},"original_publication_day":{"type":"integer","content":"13"},"average_rating":"4.00","best_book":{"type":"Book","id":{"type":"integer","content":"29093239"},"title":"Pirate (Fargo Adventure, #8)","author":{"id":{"type":"integer","content":"18411"},"name":"Clive Cussler"},"image_url":"https://images.gr-assets.com/books/1459113036m/29093239.jpg","small_image_url":"https://images.gr-assets.com/books/1459113036s/29093239.jpg"}}]}}}}}};
	  const searchResult = receivedData.query.results.GoodreadsResponse.search;
    const allResult = searchResult.results;
    const workResult = allResult && allResult.work && allResult.work.length > 0 && allResult.work.map((current) => {
      return {
        value: current['id']['content'],
        author: current['best_book']['author']['name'],
        label: current['best_book']['title'],
        bookName: current['best_book']['title'],
        URL: current['best_book']['image_url'],
        averageRating: current['average_rating'],
        id: current['id']['content']
      }
    });
	  return {
	    get: jest.fn(() => Promise.resolve(workResult)),
	  };
	});

	it ('component renders correctly', () => {
    expect(component).toMatchSnapshot();
  });

  it('fetching search data', () => {
  	const searchTerm = 'Pirate';
    const url = `${searchAPIURL}?key=${APIKEY}&q=${searchTerm}&search=title`;
    const proxyUrl = proxify(url, { inputFormat: 'xml' });
  	expect(axios.get).toHaveBeenCalled();
  });

  it('check child component AsyncSelect Present or not', () => {
    expect(component.find(AsyncSelect).length).toEqual(1);
  });

  it('calling searchHandleChange function to fetch data', () => {
  	const props = {
      onSearchCallback: jest.fn().mockReturnThis()
    };
    const wrapper = mount(<SearchComponent {...props} />);
  	const spy = jest.spyOn(wrapper.instance(), 'searchHandleChange');
  	wrapper.find(AsyncSelect).simulate('change', { currentTarget: { value: 'Pirate'}}, spy);
  	expect(spy).toBeCalled();
  });
  
})