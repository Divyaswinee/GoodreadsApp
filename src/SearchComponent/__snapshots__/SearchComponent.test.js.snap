// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchComponent component renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchComponent />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Async
        backspaceRemoves={true}
        cache={false}
        cacheOptions={false}
        className="searchBoxCustomize"
        defaultOptions={true}
        loadOptions={[Function]}
        onChange={[Function]}
        placeholder="Search Book"
        value=""
      />,
      "className": "searchContainer",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "backspaceRemoves": true,
        "cache": false,
        "cacheOptions": false,
        "className": "searchBoxCustomize",
        "defaultOptions": true,
        "loadOptions": [Function],
        "onChange": [Function],
        "placeholder": "Search Book",
        "value": "",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Async
          backspaceRemoves={true}
          cache={false}
          cacheOptions={false}
          className="searchBoxCustomize"
          defaultOptions={true}
          loadOptions={[Function]}
          onChange={[Function]}
          placeholder="Search Book"
          value=""
        />,
        "className": "searchContainer",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "backspaceRemoves": true,
          "cache": false,
          "cacheOptions": false,
          "className": "searchBoxCustomize",
          "defaultOptions": true,
          "loadOptions": [Function],
          "onChange": [Function],
          "placeholder": "Search Book",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
